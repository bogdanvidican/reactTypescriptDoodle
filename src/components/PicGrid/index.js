export { default } from './PicGrid.tsx';
