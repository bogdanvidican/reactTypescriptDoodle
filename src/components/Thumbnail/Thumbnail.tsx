import * as React from 'react';
import { WithStyles, withStyles, GridListTile, GridListTileBar, createStyles } from '@material-ui/core';

type ClassKeys = 'tile';

export interface IProps {
  title: string,
  thumb: string,
}

export interface InnerProps extends IProps, WithStyles<ClassKeys> {}

const style = () => createStyles({
  tile: {
    width: 200,
    height: 200,
    margin: 12.5,
    borderRadius: '4px',
  },
});

export const ThumbnailComponent = ({ title, thumb, classes }: InnerProps) => {
  return (
    <GridListTile classes={{ tile: classes.tile }}>
      <img src={thumb} alt={title} />
      <GridListTileBar
        title={title}
      />
    </GridListTile>
  )
};

export const Thumbnail = withStyles(style)(ThumbnailComponent);
