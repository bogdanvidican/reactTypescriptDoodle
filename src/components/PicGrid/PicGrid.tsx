import * as React from 'react';
import { WithStyles, withStyles, Theme, createStyles } from '@material-ui/core';
import IPicture from '../Cards';
import { Thumbnail } from '../Thumbnail';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import ListSubheader from '@material-ui/core/ListSubheader';

type Classes = 'root' | 'gridList' | 'title';

export interface IProps {
  type: string,
  pictures: IPicture[],
}

export interface InnerProps extends IProps, WithStyles<Classes> {}

const styles = (theme: Theme) => createStyles({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper,
  },
  gridList: {
    width: 450,
  },
  title: {
    height: 80,
  }
});

export const PicGridComponent = ({ type, pictures, classes }: InnerProps) => {
  return (
    <div className={classes.root}>
      <GridList cellHeight={200} className={classes.gridList}>
        <GridListTile cols={2} style={{height: 'auto'}}>
          <ListSubheader component="div">{type.toUpperCase()}</ListSubheader>
        </GridListTile>
        {pictures.map(pic => (
          <Thumbnail
            key={pic.thumb}
            title={pic.title}
            thumb={pic.thumb}
          />
        ))}
      </GridList>
    </div>
  );
}

export const PicGrid = withStyles(styles)(PicGridComponent);
