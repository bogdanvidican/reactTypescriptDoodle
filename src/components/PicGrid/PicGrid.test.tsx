import * as React from 'react';
import { shallow } from 'enzyme';
import { PicGridComponent } from './PicGrid';
import { Thumbnail } from '../Thumbnail';
import { InnerProps } from './PicGrid';

let wrapper: any;
const props: InnerProps = {
    type: 'liked',
    pictures: [
      {
        url: 'https://some.test.url_1',
        thumb: 'https://some.test.thumb_1',
        title: 'some test title_1',
      },
      {
        url: 'https://some.test.url_1',
        thumb: 'https://some.test.thumb_1',
        title: 'some test title_1',
      },
      {
        url: 'https://some.test.url_2',
        thumb: 'https://some.test.thumb_2',
        title: 'some test title_2',
      },
      {
        url: 'https://some.test.url_3',
        thumb: 'https://some.test.thumb_3',
        title: 'some test title_3',
      },{
        url: 'https://some.test.url_4',
        thumb: 'https://some.test.thumb_4',
        title: 'some test title_4',
      }
    ],
    classes: {
      root: 'root',
      gridList: 'gridList',
      title: 'title',
    }
};

beforeEach(() => {
  wrapper = shallow(<PicGridComponent {...props} />);
});

describe('<PicGrid />', () => {
  const nrOfPics = props.pictures.length;
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it(`should render component <Thumbnail /> ${nrOfPics} times`, () => {
    const thumbs = wrapper.find(Thumbnail).getElements();
    expect(thumbs).toHaveLength(nrOfPics);
  });
});
