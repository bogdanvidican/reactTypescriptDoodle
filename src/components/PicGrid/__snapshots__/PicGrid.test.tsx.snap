// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PicGrid /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "gridList": "gridList",
            "root": "root",
            "title": "title",
          }
    }
    pictures={
        Array [
            Object {
              "thumb": "https://some.test.thumb_1",
              "title": "some test title_1",
              "url": "https://some.test.url_1",
            },
            Object {
              "thumb": "https://some.test.thumb_1",
              "title": "some test title_1",
              "url": "https://some.test.url_1",
            },
            Object {
              "thumb": "https://some.test.thumb_2",
              "title": "some test title_2",
              "url": "https://some.test.url_2",
            },
            Object {
              "thumb": "https://some.test.thumb_3",
              "title": "some test title_3",
              "url": "https://some.test.url_3",
            },
            Object {
              "thumb": "https://some.test.thumb_4",
              "title": "some test title_4",
              "url": "https://some.test.url_4",
            },
          ]
    }
    type="liked"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(GridList)
        cellHeight={200}
        className="gridList"
>
        <WithStyles(GridListTile)
                cols={2}
                style={
                        Object {
                                "height": "auto",
                              }
                }
        >
                <WithStyles(ListSubheader)
                        component="div"
                >
                        LIKED
                </WithStyles(ListSubheader)>
        </WithStyles(GridListTile)>
        <WithStyles(Component)
                thumb="https://some.test.thumb_1"
                title="some test title_1"
        />
        <WithStyles(Component)
                thumb="https://some.test.thumb_1"
                title="some test title_1"
        />
        <WithStyles(Component)
                thumb="https://some.test.thumb_2"
                title="some test title_2"
        />
        <WithStyles(Component)
                thumb="https://some.test.thumb_3"
                title="some test title_3"
        />
        <WithStyles(Component)
                thumb="https://some.test.thumb_4"
                title="some test title_4"
        />
</WithStyles(GridList)>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cellHeight": 200,
        "children": Array [
          <WithStyles(GridListTile)
            cols={2}
            style={
                        Object {
                                    "height": "auto",
                                  }
            }
>
            <WithStyles(ListSubheader)
                        component="div"
            >
                        LIKED
            </WithStyles(ListSubheader)>
</WithStyles(GridListTile)>,
          Array [
            <WithStyles(Component)
              thumb="https://some.test.thumb_1"
              title="some test title_1"
/>,
            <WithStyles(Component)
              thumb="https://some.test.thumb_1"
              title="some test title_1"
/>,
            <WithStyles(Component)
              thumb="https://some.test.thumb_2"
              title="some test title_2"
/>,
            <WithStyles(Component)
              thumb="https://some.test.thumb_3"
              title="some test title_3"
/>,
            <WithStyles(Component)
              thumb="https://some.test.thumb_4"
              title="some test title_4"
/>,
          ],
        ],
        "className": "gridList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(ListSubheader)
              component="div"
>
              LIKED
</WithStyles(ListSubheader)>,
            "cols": 2,
            "style": Object {
              "height": "auto",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "LIKED",
              "component": "div",
            },
            "ref": null,
            "rendered": "LIKED",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "https://some.test.thumb_1",
          "nodeType": "class",
          "props": Object {
            "thumb": "https://some.test.thumb_1",
            "title": "some test title_1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "https://some.test.thumb_1",
          "nodeType": "class",
          "props": Object {
            "thumb": "https://some.test.thumb_1",
            "title": "some test title_1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "https://some.test.thumb_2",
          "nodeType": "class",
          "props": Object {
            "thumb": "https://some.test.thumb_2",
            "title": "some test title_2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "https://some.test.thumb_3",
          "nodeType": "class",
          "props": Object {
            "thumb": "https://some.test.thumb_3",
            "title": "some test title_3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "https://some.test.thumb_4",
          "nodeType": "class",
          "props": Object {
            "thumb": "https://some.test.thumb_4",
            "title": "some test title_4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(GridList)
          cellHeight={200}
          className="gridList"
>
          <WithStyles(GridListTile)
                    cols={2}
                    style={
                              Object {
                                        "height": "auto",
                                      }
                    }
          >
                    <WithStyles(ListSubheader)
                              component="div"
                    >
                              LIKED
                    </WithStyles(ListSubheader)>
          </WithStyles(GridListTile)>
          <WithStyles(Component)
                    thumb="https://some.test.thumb_1"
                    title="some test title_1"
          />
          <WithStyles(Component)
                    thumb="https://some.test.thumb_1"
                    title="some test title_1"
          />
          <WithStyles(Component)
                    thumb="https://some.test.thumb_2"
                    title="some test title_2"
          />
          <WithStyles(Component)
                    thumb="https://some.test.thumb_3"
                    title="some test title_3"
          />
          <WithStyles(Component)
                    thumb="https://some.test.thumb_4"
                    title="some test title_4"
          />
</WithStyles(GridList)>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cellHeight": 200,
          "children": Array [
            <WithStyles(GridListTile)
              cols={2}
              style={
                            Object {
                                          "height": "auto",
                                        }
              }
>
              <WithStyles(ListSubheader)
                            component="div"
              >
                            LIKED
              </WithStyles(ListSubheader)>
</WithStyles(GridListTile)>,
            Array [
              <WithStyles(Component)
                thumb="https://some.test.thumb_1"
                title="some test title_1"
/>,
              <WithStyles(Component)
                thumb="https://some.test.thumb_1"
                title="some test title_1"
/>,
              <WithStyles(Component)
                thumb="https://some.test.thumb_2"
                title="some test title_2"
/>,
              <WithStyles(Component)
                thumb="https://some.test.thumb_3"
                title="some test title_3"
/>,
              <WithStyles(Component)
                thumb="https://some.test.thumb_4"
                title="some test title_4"
/>,
            ],
          ],
          "className": "gridList",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(ListSubheader)
                component="div"
>
                LIKED
</WithStyles(ListSubheader)>,
              "cols": 2,
              "style": Object {
                "height": "auto",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "LIKED",
                "component": "div",
              },
              "ref": null,
              "rendered": "LIKED",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "https://some.test.thumb_1",
            "nodeType": "class",
            "props": Object {
              "thumb": "https://some.test.thumb_1",
              "title": "some test title_1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "https://some.test.thumb_1",
            "nodeType": "class",
            "props": Object {
              "thumb": "https://some.test.thumb_1",
              "title": "some test title_1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "https://some.test.thumb_2",
            "nodeType": "class",
            "props": Object {
              "thumb": "https://some.test.thumb_2",
              "title": "some test title_2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "https://some.test.thumb_3",
            "nodeType": "class",
            "props": Object {
              "thumb": "https://some.test.thumb_3",
              "title": "some test title_3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "https://some.test.thumb_4",
            "nodeType": "class",
            "props": Object {
              "thumb": "https://some.test.thumb_4",
              "title": "some test title_4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
